Added Story/Roleplaying Generator feature

Rhystic1 opened the pull request:
Added a Story/Roleplaying Generator feature.

This requires that the user provides a personal API key for either Gemini, Perplexity or OpenRouter (I will work on supporting more LLM API providers soon)

There are two modes: Story Mode (User provides the initial setting and characters, then AI controls all the characters and does the rest, though the user can still steer the story in the desired way), and Roleplaying Mode (the user still provides the initial setting and characters in the scene but then the chat messages are interpreted as dialogue, except when using "[]").

This leverages the existing Live2D Visualizer capabilities so that the displayed characters and animations are synchronized (including yapping) depending on the context and emotion of the characters in the scene.

Obviously it is not perfect and there is room for improvement, but so far it works pretty well, and the model intelligently understands the context of the game and the characters, especially on higher end models.

Example:
<img width="2556" height="1277" alt="image" src="https://github.com/user-attachments/assets/51f3493d-25b5-4bf3-b7e5-a26d4d7fe560" />

Automatic background selection has not been implemented yet.

Koshirei commented:
honestly it's pretty solid.
I've got to trust you for the AI part because I do not own a single token. so unless there is a token-less chatbot I won't really be of any help with this feature.
I will want a several more things.

1. I want you to mention anywhere that the API key will be stored in the user's local storage and not on any server ( similar to how I mention it for color background & custom spine loading )
2. I'd like you to write a notice/help page with a more detailed explanation on how everything works. ( without watching your video I'd have no clue what to do, I though I had to manually select chime somewhere to talk to "her" )
3. I'll need you to give me a color you'd like to be added in the credits page ( you could add yourself too )


Since I can't test the AI itself I've just done a small testing of the visualiser page to watch for any regression, and I found one with the layer editor previewing as mentionned above, if you can revert it or fix it back your own way I'd be glad ( aka I won't accept the PR if it's bugged )

I'll wait for your "GO" to pass the PR, seeing you still making frequent commits tells me it's not entirely done.
Once the PR is done it'll be your own feature. I won't bother trying to update it myself it you ever feels like stopping, but I won't delete it either like the chibis, I'll happily watch PRs and let you know if I ever get contacted about it.

Koshirei commented on src/components/views/StoryGenerator.vue:
```diff
@@ -0,0 +1,24 @@
+const market = useMarket()
+
+market.live2d.current_id = ''
+
```
add 
`onMounted(() => {
  market.load.endLoad()
})`

to make the loading bar disappear from the top of the screen

Rhystic1 commented:
Thank you for reviewing this!

Yes, I'm going to address all the points that you've mentioned, including adding a small guide for people to clarify how to use the feature.

As for testing it, you could get a free Gemini API key from Google AI Studio which will allow you to use the Gemini 2.5 Flash model with pretty generous limits. If I remember correctly, you don't even have to insert any billing info. I'm on my phone now but I'll send you the link when I'm home. 

Otherwise, don't worry about it and I'll be gladly take care of maintaining this feature!

Regarding the commits - yeah, I may create one or two more before giving you the go ahead. I want to try incorporating backgrounds as well to make the section feel more polished and interactive. If it gets too complicated, I'll leave it as it is at least for the moment. I have also taken care of all the bugs I could find so far. 

Finally, sorry about that regression! I must have missed it - I will definitely fix it as soon as I can!

Koshirei commented:
For some reason Spine is loading a couple Layers/attachment by default with different values than 1/1/1/1 for r/g/b/a.
When I trigger the layer editing preview I'm doing a backup of every attachments, in theory all at 1/1/1/1 but it's not really the case.
When the preview ends I reapply all the backed up values to the spine.

Let's take Rapi's face because it's easy to reproduce

1. Spine load the sprite with 1/1/1/1 colors
2. for some reason the JS object have Rapi's face layer at 0/0.8/0.5/1 ( that was an example value but it really happens ) 
3. I backup this wrong layer color during preview
4. I reapply it after preview
5. Rapi's face now turned green due to spine's wrong default color instanciation


I don't understand why the bit that was fixing this was removed in your PR and I doubt it impacts your own feature

https://github.com/user-attachments/assets/3034ccf6-5789-4681-9fb8-38575977e896

Rhystic1 commented:
@Koshirei I believe I have fixed the regression in the last commit - please let me know if you find other issues.

Meanwhile, I have added the warnings we discussed about the API key and I have created a help popup that appears the first time a user enters the page and it can also be read again at any time by pressing on the "?" button. 

As for the backgrounds, I realized that those work very differently than the animations. This would be much more complicated to handle, so I am not going to focus on that for now. However, I will be focusing on the save/load functionality, after which I will feel confident to give you the OK to merge.

Credits-wise - if you can add me with a light-blue colour that would be great! I appreciate it!

Rhystic1 commented:
OK, I feel confident now that unless you find other regressions or you have any other doubts, it can be merged and released. I have also added a Save/Load function which I felt was necessary, in addition to a few more QoL things. 

Do you want me to add myself to the credits or do you want to handle that? I'm fine either way!
